Linking common symbols on macOS can fail

`m.f90`:
```
subroutine touch()
integer executables
common /blk/ executables
executables = executables + 1
end
```

`main.c`:
```
#include <stdio.h>

extern int struct_instance_blk;

int main(void) {
  printf("%d\n", struct_instance_blk);
  return 0;
}
```


And:
```console
$ lfortran --separate-compilation -c m.f90 -o m.o
$ ar rcs libm.a m.o
$ nm libm.a

m.o:
0000000000000004 C _struct_instance_blk
0000000000000000 T _touch
0000000000000000 t ltmp0
0000000000000018 s ltmp1
$ lfortran -o a_lfortran_conda.out main.o libm.a
Undefined symbols for architecture arm64:
  "_struct_instance_blk", referenced from:
      _main in main.o
ld: symbol(s) not found for architecture arm64
$ clang -o a_apple.out main.c libm.a
Undefined symbols for architecture arm64:
  "_struct_instance_blk", referenced from:
      _main in main-f9b441.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

This `clang` must be from the `gfortran` conda env. The native Apple Clang compiles and runs the code just fine.

LFortran should avoid relying on COMMON-style emission for these globals (emit strong BSS/data definitions instead).


----

Better repro:


```
$ cat m.f90
module m
  implicit none
  integer :: executables
contains
  subroutine touch()
    executables = executables + 1
  end subroutine touch
end module m
$ cat main.f90
program main
  use m, only: executables
  print *, executables
end program main
```

```console
$ lfortran --separate-compilation -c m.f90 -o m.o
$ lfortran --separate-compilation -I . -c main.f90 -o main.o
$ ar rcs libm.a m.o
$ lfortran  -o a.out main.o libm.a
Undefined symbols for architecture arm64:
  "___module_m_executables", referenced from:
      _main in main.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The command 'clang -o a.out main.o libm.a  -L"/Users/ondrej/repos/lfortran/src/bin/../runtime" -Wl,-rpath,"/Users/ondrej/repos/lfortran/src/bin/../runtime" -llfortran_runtime -lm' failed.
```


In this repro, LFortran is emitting the module variable as a COMMON/tentative
  symbol instead of a real definition.

  You can see it directly:

  - main.o: U ___module_m_executables (undefined reference)
  - m.o: C ___module_m_executables (COMMON symbol)

  That C is the problem. With some linkers (such as conda ld64), archive member
  extraction does not reliably treat that as a normal defining symbol, so
  libm.a(m.o) is not pulled in, and link fails.

  What LFortran should do instead:

  1. In the module-defining object (m.o), emit ___module_m_executables as a real
     global data definition:

  - B (zero-initialized in .bss) or D (initialized in .data), not C.

  2. In users (main.o), keep it as an external reference (U), which is already
     fine.

  So the fix is in LFortran codegen/ABI lowering for module variables in
  separate compilation: stop lowering them via COMMON-style emission, and emit a
  strong global definition in the module object. That makes archive linking
  robust across linkers.

--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -3173,6 +3173,7 @@ RUN(NAME separate_compilation_25 LABELS gfortran llvm_submodule EXTRAFILES separ
 RUN(NAME separate_compilation_26 LABELS gfortran llvm_submodule EXTRAFILES separate_compilation_26a.f90 separate_compilation_26b.f90)
 RUN(NAME separate_compilation_27 LABELS gfortran llvm_submodule EXTRAFILES separate_compilation_27a.f90 separate_compilation_27b.f90)
 RUN(NAME separate_compilation_28 LABELS gfortran llvm EXTRAFILES separate_compilation_28a.f90 separate_compilation_28b.f90 EXTRA_ARGS --implicit-interface)
+RUN(NAME separate_compilation_29 LABELS gfortran llvm EXTRAFILES separate_compilation_29a.f90 separate_compilation_29b.f90 EXTRA_ARGS --separate-compilation)
 
 
 RUN(NAME no_explicit_return_type_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc

--- a/integration_tests/separate_compilation_29.f90
+++ b/integration_tests/separate_compilation_29.f90
@@ -1,0 +1,11 @@
+program separate_compilation_29
+    use separate_compilation_29a_module, only: sa, get_a
+    use separate_compilation_29b_module, only: sb, get_b
+    implicit none
+
+    call sa()
+    call sb()
+
+    if (get_a() /= 1) error stop 1
+    if (get_b() /= 2) error stop 2
+end program separate_compilation_29

--- a/integration_tests/separate_compilation_29a.f90
+++ b/integration_tests/separate_compilation_29a.f90
@@ -1,0 +1,20 @@
+module separate_compilation_29a_module
+    implicit none
+
+    type :: t
+        integer :: x
+    end type t
+
+    type(t) :: targets
+
+contains
+
+    subroutine sa()
+        targets%x = 1
+    end subroutine sa
+
+    integer function get_a()
+        get_a = targets%x
+    end function get_a
+
+end module separate_compilation_29a_module

--- a/integration_tests/separate_compilation_29b.f90
+++ b/integration_tests/separate_compilation_29b.f90
@@ -1,0 +1,20 @@
+module separate_compilation_29b_module
+    implicit none
+
+    type :: t
+        integer :: x
+    end type t
+
+    type(t) :: targets
+
+contains
+
+    subroutine sb()
+        targets%x = 2
+    end subroutine sb
+
+    integer function get_b()
+        get_b = targets%x
+    end function get_b
+
+end module separate_compilation_29b_module

--- a/src/libasr/codegen/asr_to_llvm.cpp
+++ b/src/libasr/codegen/asr_to_llvm.cpp
@@ -4310,11 +4310,7 @@ public:
             // for external global variable with bindc, use the C name
             llvm_var_name = x.m_bindc_name;
         } else {
-            if ( !( ASRUtils::is_struct(*x.m_type) || ASRUtils::is_class_type(x.m_type) ) ) {
-                llvm_var_name = mangle_prefix + x.m_name;
-            } else {
-                llvm_var_name = x.m_name;
-            }
+            llvm_var_name = mangle_prefix + x.m_name;
         }
         if (x.m_type->type == ASR::ttypeType::Integer
             || x.m_type->type == ASR::ttypeType::UnsignedInteger) {

